acnportal/acnsim/network/charging_network.py

```diff
@@ -327,6 +329,59 @@ def is_feasible(self, schedule_matrix, linear=False, violation_tolerance=None, r
         return np.all(np.tile(self.magnitudes + np.maximum(violation_tolerance, rel_magnitude_tol), (schedule_length, 1)).T >= np.abs(aggregate_currents))
 
 
+    def to_dict(self, context_dict=None):
+        """ Implements BaseSimObj.to_dict. """
+        context_dict, = base.none_to_empty_dict(context_dict)
+        args_dict = {}
+
+        # Serialize non-nested attributes.
+        nn_attr_lst = ['violation_tolerance', 'relative_tolerance']
+        for attr in nn_attr_lst:
+            args_dict[attr] = getattr(self, attr)
+
+        args_dict['_EVSEs'] = {
+            station_id : evse.to_registry(context_dict=context_dict)['id']
+            for station_id, evse in self._EVSEs.items()
+        }
+
+        args_dict['constraint_matrix'] = self.constraint_matrix.tolist()
+        args_dict['magnitudes'] = self.magnitudes.tolist()
+        args_dict['_voltages'] = self._voltages.tolist()
+        args_dict['_phase_angles'] = self._phase_angles.tolist()
+
+        args_dict['constraint_index'] = self.constraint_index
+
+        return args_dict
+
+    @classmethod
+    def from_dict(cls, in_dict, context_dict=None, loaded_dict=None,
+                  cls_kwargs=None):
+        """ Implements BaseSimObj.from_dict. """
+        context_dict, loaded_dict, cls_kwargs = \
+            base.none_to_empty_dict(context_dict, loaded_dict, cls_kwargs)
+
+        out_obj = cls(violation_tolerance=in_dict['violation_tolerance'],
+                      relative_tolerance=in_dict['relative_tolerance'],
+                      **cls_kwargs)
+
+        out_obj._EVSEs = {
+            station_id : base.read_from_id(
+                evse, context_dict=context_dict, loaded_dict=loaded_dict
+            )
+            for station_id, evse in in_dict['_EVSEs'].items()
+        }
+
+        out_obj.constraint_matrix = \
+            np.array(in_dict['constraint_matrix'])
+        out_obj.magnitudes = \
+            np.array(in_dict['magnitudes'])
+
+        out_obj.constraint_index = in_dict['constraint_index']
+        out_obj._voltages = np.array(in_dict['_voltages'])
+        out_obj._phase_angles = np.array(in_dict['_phase_angles'])
```

Comment:
Keep order between to_dict and from_dict consistent.

Reply:
Where it was reasonable to do so, I reordered to_dict and from_dict attribute handling such that the order in which the attributes were processed was the same for both serialization and deserialization.